tests/new_api/test_device.py

```diff
@@ -649,6 +677,11 @@ def test_derivatives_single_expval(
         self, theta, phi, dev, obs, execute_and_derivatives, batch_obs
     ):
         """Test that the jacobian is correct when a tape has a single expectation value"""
+        if isinstance(obs, qml.SparseHamiltonian) and dev.dtype == np.complex64:
```

Comment:
Is this for all backends or only for `lightning.gpu`?

Reply:
[Original post](https://github.com/PennyLaneAI/pennylane-lightning/pull/908#discussion_r1759155951)
This limitation comes from a hard code in Pennylane [ref](https://github.com/PennyLaneAI/pennylane/blob/060bb9a7479dcf6735671d809da19d500c4b7a89/pennylane/ops/qubit/hamiltonian.py#L522)